### PR TITLE
Notification Settings: Remove sites-list

### PR DESCRIPTION
**Pull request by rodrigoi**
This PR removes `sites-list` and `UserStore` from Notification Settings (`me/notification-settings`), and adds minor changes to some `connect` exports.

To test:

- check out this branch
- go to http://calypso.localhost:3000/me/notifications
- verify that settings are saved and loaded as expected
- verify that the list of sites is complete
- verify that the primary site is displayed first
- verify it works correctly with a clean state.
- no other tabs under Notification Settings should be affected.

Part of #13279

### CODE REVIEW

File: client/me/notification-settings/blogs-settings/blog.jsx [hunk 1]
```diff
@@ -1,40 +1,53 @@
...
 		disableToggle: PropTypes.bool,
 		settings: PropTypes.instanceOf( Immutable.Map ).isRequired,
 		hasUnsavedChanges: PropTypes.bool.isRequired,
 		onToggle: PropTypes.func.isRequired,
 		onSave: PropTypes.func.isRequired,
 		onSaveToAll: PropTypes.func.isRequired
-	},
+	};
+
+	state = { isExpanded: false	};
```
Comment (tyxla) [inline client/me/notification-settings/blogs-settings/blog.jsx]: There's a tab here that should be a space instead.

File: client/me/notification-settings/blogs-settings/header.jsx [hunk 2]
```diff
@@ -63,29 +55,33 @@ export default React.createClass( {
...
+				<div className="blogs-settings__header-legend">
 					<em>{ this.getLegend() }</em>
 				</div>
-				{ ! this.props.disableToggle ?
-				<div className="notification-settings-blog-settings-header__expand">
-					<a className={ 'noticon noticon-' + ( this.state.isExpanded ? 'collapse' : 'expand' ) }></a>
-				</div> :
-				null }
+				{ ! this.props.disableToggle
+					? <div className="blogs-settings__header-expand">
```
Comment (tyxla) [inline client/me/notification-settings/blogs-settings/header.jsx]: Formatting appears a bit off here. We can wrap the `<div>` in a set of parens, and keep the opening `<div>` on a new line for consistency:

```
? (
	<div className="blogs-settings__header-expand">
		<a className={ 'noticon noticon-' + ( this.state.isExpanded ? 'collapse' : 'expand' ) }></a>
	</div>
)
```

File: client/me/notification-settings/blogs-settings/index.jsx [hunk 3]
```diff
@@ -1,74 +1,97 @@
...
 
 	render() {
-		if ( ! this.props.blogs.initialized || ! this.props.devices.initialized || ! this.props.settings ) {
-			return <Placeholder />;
+		const { sites, translate } = this.props;
+
+		if ( ! sites || ! this.props.devices.initialized || ! this.props.settings ) {
+			return (
+				<div>
+					<QuerySites allSites />
```
Comment (tyxla) [inline client/me/notification-settings/blogs-settings/index.jsx]: Is this one necessary? #13094 introduced a `<QuerySites allSites />` instance already. If you remove this one, the `<div>` can also be removed.

File: client/me/notification-settings/blogs-settings/blog.jsx (hunk 1 above)
Comment (rodrigoi) [inline client/me/notification-settings/blogs-settings/blog.jsx]: Thanks! Fixed on ec37308205f7750985ed2f37c0537a0b067c1457

File: client/me/notification-settings/blogs-settings/header.jsx (hunk 2 above)
Comment (rodrigoi) [inline client/me/notification-settings/blogs-settings/header.jsx]: Fixed on 657b16489392a6e85e877cce138cb11841da9084

File: client/me/notification-settings/blogs-settings/index.jsx (hunk 3 above)
Comment (rodrigoi) [inline client/me/notification-settings/blogs-settings/index.jsx]: Absolutely right. This is not necessary. I must have had a weird state on my browser. Fixed on bb80c9cfd49913748d417d2974508abc8cd26668

**Comment by rodrigoi** [comment]
@tyxla the primary site problem is indeed a regression caused most likely by the switch from `map` on `sites-list` to `reduce` on the sites reducer. Looks like lodash iterates the index keys in ascending order.
I've aded a small fix using `getPrimarySiteId`, but we should add a note to `getSites` to specify that the order of items returned from the server may not be preserved.
cc: @mtias

**Pull request merged**